packages/plugin-commands-publishing/src/runNpm.ts

```diff
@@ -0,0 +1,17 @@
+import { packageManager } from '@pnpm/cli-utils'
+import getConfig, { types as allTypes } from '@pnpm/config'
+import runNpm from '@pnpm/run-npm'
+import R = require('ramda')
+
+export default async function run (args: string[]) {
+  const { config } = await getConfig({
+    cliOptions: {},
+    packageManager,
+    rcOptionsTypes: {
+      ...R.pick([
+        'npm-path',
+      ], allTypes),
+    },
+  })
+  return runNpm(config.npmPath, args)
+}
```

Comment:
This file is not needed. The configs are read on each command execution, so you just repeat what already was done.

You should update the following line to contain this npm-path:

https://github.com/pnpm/pnpm/blob/a56c7856a2ead878bc4dc4285514da9b1e84175c/packages/plugin-commands-publishing/src/publish.ts#L17

So something like

```ts
export function rcOptionsTypes () {
return {
  ...cliOptionsTypes(),
  ...R.pick([
      'npm-path',
    ], allTypes)
}
```

And here you should add `'npm-path' |`:

https://github.com/pnpm/pnpm/blob/a56c7856a2ead878bc4dc4285514da9b1e84175c/packages/plugin-commands-publishing/src/recursivePublish.ts#L31

That way, `npmPath` will be in the opts passed to the publishing plugin.

Same changes should be done in other plugins that run npm